### PR TITLE
fix #208

### LINKED ISSUE #208: issues with multi entity notification (wrong servicepath created)
**Describe the bug**
When a multiple notification is sent, the service path created is a comma separated version of all servicepaths of the entities.
It could be that the header sent by orion is exactly that. We should then split it and attach it one for entity.

entity_id | entity_type | fiware_servicepath | powerstate | time_index | timeinstant
-- | -- | -- | -- | -- | --
urn:ngsi-ld:Streetlight:sl-6 | Streetlight | /Diorama,/Diorama,/Diorama,/Diorama,/Diorama,/Diorama | NULL | 1560425955297 (2019-06-13T11:39:15.297Z) | NULL
urn:ngsi-ld:Streetlight:sl-1 | Streetlight | /Diorama,/Diorama,/Diorama,/Diorama,/Diorama,/Diorama | NULL | 1560422941000 (2019-06-13T10:49:01.000Z) | 1560422941000 (2019-06-13T10:49:01.000Z)
urn:ngsi-ld:Streetlight:sl-3 | Streetlight | /Diorama,/Diorama,/Diorama,/Diorama,/Diorama,/Diorama | NULL | 1560422941000 (2019-06-13T10:49:01.000Z) | 1560422941000 (2019-06-13T10:49:01.000Z)
urn:ngsi-ld:Streetlight:sl-2 | Streetlight | /Diorama,/Diorama,/Diorama,/Diorama,/Diorama,/Diorama | NULL | 1560422941000 (2019-06-13T10:49:01.000Z) | 1560422941000 (2019-06-13T10:49:01.000Z)
urn:ngsi-ld:Streetlight:sl-5 | Streetlight | /Diorama,/Diorama,/Diorama,/Diorama,/Diorama,/Diorama | NULL | 1560425955297 (2019-06-13T11:39:15.297Z) | NULL
urn:ngsi-ld:Streetlight:sl-4 | Streetlight | /Diorama,/Diorama,/Diorama,/Diorama,/Diorama,/Diorama | NULL | 1560360355000 (2019-06-12T17:25:55.000Z) | 1560360355000 (2019-06-12T17:25:55.000Z)

**Environment (please complete the following information):**
 - CrateDB 3.3.3

### PR DESCRIPTION
**Pull request by chicco785**
* added code to group entities by servicepath when multiple notifications arrive at once (#208)
* add tests and integration tests (so we have also test with orion triggering such case)
* fix notification tests to rely only on api calls (remove call to translator)
* changed error code to 400 when due to invalid request
* upgraded python to 3.8.5
* fix integration test to use new docker image with entrypoint (#362)

### CODE REVIEW

File: src/translators/sql_translator.py [hunk 1]
```diff
@@ -170,16 +170,46 @@ def insert(self, entities, fiware_service=None, fiware_servicepath=None):
...
+                    res = self._insert_entities_of_type(et,
+                                                        entities_by_type[et],
+                                                        fiware_service,
+                                                        path)
+        else:
+            msg = 'Multiple servicePath are allowed only ' \
+                  'if their size is match the size of entities'
+            raise InvalidHeaderValue('Fiware-ServicePath',
+                                     fiware_servicepath, msg)
+
```
Comment (c0c0n3) [inline src/translators/sql_translator.py:212]: what happens if there's only one entity and an empty service path?

    service_paths = ['']
    for idx, path in enumerate(service_paths):
        print(f"{idx}: {path}")

    >>>     0:

so I think we'd be inserting the entity with an empty path? In general, do we still allow clients not to specify a service path, so e.g. a notification w/ 3 entities and no service path results in all three entities still being inserted under the empty path?

File: src/tests/run_load_tests.sh [hunk 2]
```diff
@@ -1,7 +1,5 @@
 #!/usr/bin/env bash
 
-docker build --cache-from smartsdk/quantumleap -t smartsdk/quantumleap ../../
```
Comment (c0c0n3) [inline src/tests/run_load_tests.sh:3]: don't we need this line? I think w/o it, when you run the script you get the latest image in the docker registry which doesn't not necessarily reflects the latest code changes? for example, say I check out a branch someone is working on, then run this script. The tests will run with my local image, which won't have the changes in the branch?

File: src/translators/sql_translator.py (hunk 1 above)
Comment (chicco785) [reply src/translators/sql_translator.py:212]: may the correct one is `service_paths = [None]` 
all actually boils down to what orion provide us, right?

File: src/tests/run_load_tests.sh (hunk 2 above)
Comment (chicco785) [reply src/tests/run_load_tests.sh:3]: correct, but for load test i thought it was not needed. can be inserted back

File: src/translators/sql_translator.py (hunk 1 above)
Comment (chicco785) [reply src/translators/sql_translator.py:212]: see line 178, so if no service path, we exactly create an array with `service_paths = [None]`

Comment (c0c0n3) [reply src/translators/sql_translator.py:212]: yep, line 178 is what makes a list with one entry if there's no service path (`None` or `''`). just wanted to double check this was intentional.

> boils down to what orion provide us, right?

yep, you're right. I don't think we've got much of a choice here. If the notification contains no service path probably we should still insert the entity. Looking at the code I think we do that in the case of one entity, but what does Orion do when notifying multiple entities with no service path? Will the notification contain a list of empty service paths, e.g. for 3 entities

    path = ' , , '
    entities = [e1, e2, e3]

or no path at all, e.g.

    path = ''
    entities = [e1, e2, e3]

If I read the code correctly, in the first case we insert all 3 entities under the empty path whereas in the second case we raise an `InvalidHeaderValue` exception. If this is the case, then my question is: do we have to insert entities if there's no service path or it is empty?

File: src/tests/run_load_tests.sh (hunk 2 above)
Comment (c0c0n3) [reply src/tests/run_load_tests.sh:3]: cool, we could put it back in after rebasing #347 on master? or is it easier to do in this PR?

File: src/translators/sql_translator.py (hunk 1 above)
Comment (chicco785) [reply src/translators/sql_translator.py:212]: Multiple "" will work if they are in the same number as entities.

**Review by c0c0n3 — APPROVED** [review]

**Pull request merged**